### PR TITLE
docs: add missing resource tag associated with spot

### PR DESCRIPTION
**Pull request by useradd-temp**
Fixes #N/A 

**Description**

I added a resource for `KarpenterControllerPolicy` that is present in `cloudformation.yaml` in the [docs](https://github.com/aws/karpenter-provider-aws/blob/main/website/content/en/docs/getting-started/getting-started-with-karpenter/cloudformation.yaml) and [preview](https://github.com/aws/karpenter-provider-aws/blob/main/website/content/en/preview/getting-started/getting-started-with-karpenter/cloudformation.yaml), but missing from the reference in the docs.

There may be other missing, but in this PR I only added the missing `"arn:${AWS::Partition}:ec2:${AWS::Region}:*:spot-instances-request/*`.

If create a policy with this resource missing, karpenter will not be able to create a spot instance.

And conversely, I deleted resources that only existed in reference (sid : `AllowScopedEC2InstanceActions`)

If this resource is deleted, there is no problem creating a spot instance (it exists in `AllowScopedEC2InstanceActionsWithTags`).

**How was this change tested?**

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: website/content/en/docs/reference/cloudformation.md [hunk 1]
```diff
@@ -113,7 +113,7 @@ Someone wanting to add Karpenter to an existing cluster, instead of using `cloud
 
 The AllowScopedEC2InstanceActions statement ID (Sid) identifies a set of EC2 resources that are allowed to be accessed with
 [RunInstances](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RunInstances.html) and [CreateFleet](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateFleet.html) actions.
-For `RunInstances` and `CreateFleet` actions, the Karpenter controller can read (but not create) `image`, `snapshot`, `spot-instances-request`, `security-group`, `subnet` and `launch-template` EC2 resources, scoped for the particular AWS partition and region.
+For `RunInstances` and `CreateFleet` actions, the Karpenter controller can read (but not create) `image`, `snapshot`, `security-group`, `subnet` and `launch-template` EC2 resources, scoped for the particular AWS partition and region.
```
Comment (jonathan-innis) [inline website/content/en/docs/reference/cloudformation.md:116]: Can you add this change to each version of the docs? `/preview/` and `/v0.33/` paths as well?

**Comment by jonathan-innis** [comment]
Nice catch 🎉

File: website/content/en/docs/reference/cloudformation.md (hunk 1 above)
Comment (useradd-temp) [reply website/content/en/docs/reference/cloudformation.md:116]: fixed it. And I removed unnecessary whitespace that was only in `/preview/` but not in `/docs/`. 
(`/v0.31/` does not have the same issue)

**Comment by useradd-temp** [comment]
Additionally, removed unnecessary whitespace that was only in `/preview/` but not in `/docs/`. 
(`/v0.31/` does not have the same issue)

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Pull request merged**